compiler/lib/targetint.ml

```diff
@@ -0,0 +1,138 @@
+type t = Int32.t
+
+let num_bits_ = ref 0
+
+let set_num_bits x = num_bits_ := x
+
+let num_bits () =
+  match !num_bits_ with
+  | (31 | 32) as x -> x
+  | x -> failwith (Printf.sprintf "Targetint.num_bits %d unsupported" x)
```

Comment:
My first reaction was: “Why hasn’t he used a module chosen according to the compilation target?” But then I understood that this information is not available at initialization phase. Interesting.

Reply:
We could do it at initialization phase by linking different libraries. But then, you cannot change your mind at runtime